src/App.js

```diff
@@ -453,14 +469,43 @@ const App = () => {
 
       resp["ranges"] = ranges;
       setQcData(resp);
-      setQcDims(`${resp.retained}`);
       setShowQCLoader(false);
+    } else if (payload.type === "adt_quality_control_DATA") {
+      const { resp } = payload;
+
+      var ranges = {}, data = resp["data"], all = {};
+
+      for (const [group, gvals] of Object.entries(data)) {
+        for (const [key, val] of Object.entries(gvals)) {
+          if (!all[key]) all[key] = [Infinity, -Infinity];
+          let [min, max] = getMinMax(val);
+          if (min < all[key][0]) all[key][0] = min;
+          if (max > all[key][1]) all[key][1] = max;
+        }
+        ranges[group] = all;
+      }
+
+      resp["ranges"] = ranges;
+
+      let prevQC = {...qcData};
+      prevQC["data"]["adt_default"] = resp["data"]["default"];
```

Comment:
I assume that this is just placeholder code, given that it's pretty fragile because there's no guarantee that we even have a batch called `default`. It seems like it should be possible to handle the RNA and ADT QC events separately, given that they map to different plots; there shouldn't be a need to stuff it into the same `qcData` object.